cpp/src/barretenberg/stdlib/primitives/group/group.hpp

```diff
@@ -39,8 +42,18 @@ auto group<ComposerContext>::fixed_base_scalar_mul_g1(const field_t<ComposerCont
 
 template <typename ComposerContext>
 template <size_t num_bits>
-auto group<ComposerContext>::fixed_base_scalar_mul(const field_t<ComposerContext>& in, const size_t generator_index)
+point<ComposerContext> group<ComposerContext>::fixed_base_scalar_mul(const field_t<ComposerContext>& in,
+                                                                     const size_t generator_index)
 {
+    if constexpr (IsSimulator<ComposerContext>) {
+        if (in.get_value() == barretenberg::fr(0)) {
+            in.context->failure("input scalar to fixed_base_scalar_mul_internal cannot be 0");
+            return { 0, 0 }; // WORKTODO: std::optional?
```

Comment:
TODO: delete comment, doesn't make sense. Also choose nonzero values for return?